firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsController.java

```diff
@@ -437,6 +439,24 @@ void setCustomKey(String key, String value) {
     cacheKeyData(userMetadata.getCustomKeys());
   }
 
+  void setCustomKeys(Map<String, String> keysAndValues) {
+    // Write all the key/value pairs before doing anything computationally expensive.
+    for (Map.Entry<String, String> entry : keysAndValues.entrySet()) {
+      try {
+        userMetadata.setCustomKey(entry.getKey(), entry.getValue());
+      } catch (IllegalArgumentException ex) {
+        if (context != null && CommonUtils.isAppDebuggable(context)) {
+          throw ex;
+        } else {
+          Logger.getLogger().e("Attempting to set custom attribute with null key, ignoring.");
+          return;
```

Comment:
would you want to `continue` to write the rest? this cuts out the for loop at the problem